controller/src/test/java/io/pravega/controller/eventProcessor/impl/SerializedRequestHandlerTest.java

```diff
@@ -226,6 +229,62 @@ public boolean toPostpone(TestEvent event, long pickupTime, Throwable exception)
         stop.set(true);
     }
 
+    @Test(timeout = 10000)
+    public void testThrowsSynchronousException() throws InterruptedException, ExecutionException {
```

Comment:
I don't understand the synchronous exception issue that we are supposedly fixing in this PR and testing here. Could you please elaborate?

Reply:
Lets say i call a method that returns a future.. and that future could pass or fail. 
But if the method throws an exception rather than return a future that fails. For example, a precondition check in the method fails before it creates the exception. 

Example:
```
CompletableFuture<Void> decrementByOne(int input) {
if (input <= 0) throw new RuntimeException("value should be greater than 1");
return new CompletableFuture.completedFuture(input - 1);
}

// following will get a future back
val x = incrementByOne(100);

// following will fail synchronously and 
val x = incrementByOne(-1);
```